### PR TITLE
Add support for Docker/Windows spec

### PR DESCRIPTION
**Pull request by MisterDA**
The commit setting `safe.directory` shall be removed when our base images start shipping with
- https://github.com/ocurrent/ocaml-dockerfile/pull/108

### CODE REVIEW

File: lib/btrfs_store.ml [hunk 1]
```diff
@@ -165,8 +165,11 @@ let cache ~user t name : (string * (unit -> unit Lwt.t)) Lwt.t =
   (* Create writeable clone. *)
   let gen = cache.gen in
   Btrfs.subvolume_snapshot `RW ~src:snapshot tmp >>= fun () ->
-  let { Obuilder_spec.uid; gid } = user in
-  Os.sudo ["chown"; Printf.sprintf "%d:%d" uid gid; tmp] >>= fun () ->
+  begin match user with
+    | `Unix { Obuilder_spec.uid; gid } ->
+      Os.sudo ["chown"; Printf.sprintf "%d:%d" uid gid; tmp]
+    | _ -> assert false
```
Comment (tmcgilchrist) [inline lib/btrfs_store.ml:171]:
```suggestion
    | `Windows -> assert false (* Btfs not supported on Windows*)
```
Can this use exhaustive pattern matches here?

File: lib/rsync_store.ml [hunk 2]
```diff
@@ -144,7 +144,7 @@ let cache ~user t name =
   end >>= fun () ->
   (* Create writeable clone. *)
   let gen = cache.gen in
-  let { Obuilder_spec.uid; gid } = user in
+  let { Obuilder_spec.uid; gid } = match user with `Unix user -> user | _ -> assert false in
```
Comment (tmcgilchrist) [inline lib/rsync_store.ml:147]: Same suggestion here, please exhaustively pattern match.
rsync could work on Windows it is not implemented yet.

File: lib/runc_sandbox.ml [hunk 3]
```diff
@@ -108,7 +108,7 @@ module Json_config = struct
 
   let make {Config.cwd; argv; hostname; user; env; mounts; network; mount_secrets} t ~config_dir ~results_dir : Yojson.Safe.t =
     let user =
-      let { Obuilder_spec.uid; gid } = user in
+      let { Obuilder_spec.uid; gid } = match user with `Unix user -> user | _ -> assert false in
```
Comment (tmcgilchrist) [inline lib/runc_sandbox.ml:111]:
```suggestion
      let { Obuilder_spec.uid; gid } = match user with `Unix user -> user | `Windows -> assert false in
```

**Comment by MisterDA** [comment]
Applied your suggestions, thanks.

**Review by tmcgilchrist — APPROVED** [review]

**Comment by MisterDA** [comment]
Updated the base Windows image, good to go.

File: lib/zfs_store.ml [hunk 4]
```diff
@@ -88,11 +88,11 @@ end = struct
     else fn ()
 end
 
-let user = { Obuilder_spec.uid = Unix.getuid (); gid = Unix.getgid () }
+let user = `Unix { Obuilder_spec.uid = Unix.getuid (); gid = Unix.getgid () }
 
 module Zfs = struct
   let chown ~user t ds =
-    let { Obuilder_spec.uid; gid } = user in
+    let { Obuilder_spec.uid; gid } = match user with `Unix user -> user | _ -> assert false in
```
[the remark this answers is not in the document]
Comment (MisterDA) [reply lib/zfs_store.ml]: Thanks.

**Review by tmcgilchrist — APPROVED** [review]

**Comment by MisterDA** [comment]
These amended patches finally fix the spec tests on Windows. Please consider for merging.

**Review by tmcgilchrist — APPROVED** [review]

**Pull request merged**